src/cplus_plugin/models/constant_raster.py

```diff
@@ -0,0 +1,432 @@
+
+from __future__ import annotations
+import os
+import dataclasses
+import typing
+from qgis.core import QgsRasterLayer
+
+from .base import LayerModelComponent, ModelComponentType
+from ..definitions.constants import (
+    PATH_ATTRIBUTE,
+)
+
+
+@dataclasses.dataclass
+class ConstantRasterInfo:
+    """Value information for a constant raster.
+
+    This dataclass only handles value information (normalized and absolute values),
+    File/layer information should be handled separately by ConstantRasterComponent.
+    """
+
+    normalized: float = 0.0  # Normalized value (0-1 range)
+    absolute: float = 0.0  # Absolute constant value for creating rasters
+
+    def to_dict(self) -> dict:
+        """Serialize to dictionary."""
+        return {
+            "normalized": self.normalized,
+            "absolute": self.absolute,
+        }
+
+    @staticmethod
+    def from_dict(d: dict) -> "ConstantRasterInfo":
+        """Deserialize from dictionary."""
+        return ConstantRasterInfo(
+            normalized=float(d.get("normalized", 0.0)),
+            absolute=float(d.get("absolute", 0.0)),
+        )
+
+
+@dataclasses.dataclass
+class ConstantRasterComponent:
+    """Individual constant raster with layer model component associations.
+
+    This class links the ConstantRasterInfo with the specific layer model
+    component (NcsPathway or Activity).
+    Activities are stored by calling their type.
+    """
+
+    value_info: ConstantRasterInfo
+    component: LayerModelComponent  # Either NcsPathway or Activity
+    prefix: str = ""  # Prefix for naming
+    base_name: str = ""  # Base component name
+    suffix: str = ""  # Suffix for naming
+    alias_name: str = ""
+    path: str = ""  # Note: Returns False for components (not a file path)
+    skip_raster: bool = True  # Skip raster creation for this component (default: True)
+    enabled: bool = True  # Whether this component is enabled
+    component_id: str = ""
+    component_type: ModelComponentType = ModelComponentType.UNKNOWN
+    qgis_map_layer: typing.Optional[QgsRasterLayer] = None
+
+    def identifier(self) -> str:
+        """Generate unique identifier by concatenating naming components.
+
+        Uses prefix, base_name, and suffix to create the identifier.
+        For example: "prefix_base_name_suffix" or just components that exist.
+        """
+        parts = []
+        if self.prefix:
+            parts.append(self.prefix)
+        if self.base_name:
+            parts.append(self.base_name)
+        elif self.component:
+            parts.append(getattr(self.component, "name", ""))
+        if self.suffix:
+            parts.append(self.suffix)
+
+        if parts:
+            return "_".join(parts)
+        return self.alias_name or self.component_id
+
+    def to_dict(self) -> dict:
+        """Serialize to dictionary."""
+        return {
+            "value_info": self.value_info.to_dict() if self.value_info else {},
```

Comment:
@osundwajeff use constant values either in constants.py or defaults.py in the definitions module. Look at the pattern applied in other modules.